v3/src/components/graph/models/graph-controller.ts

```diff
@@ -80,8 +81,9 @@ export class GraphController {
     }
   }
 
-  handleAttributeAssignment(graphPlace: GraphPlace, attrID: string) {
-    const {graphModel, layout, dataset} = this,
+  handleAttributeAssignment(graphPlace: GraphPlace, dataSetID: string, attrID: string) {
+    const {graphModel, layout} = this,
+      dataset = getDataSetFromId(graphModel, dataSetID),
       dataConfig = graphModel?.config
```

Comment:
If the incoming dataset is different than this.dataset we need to clear existing attributes and start fresh.

Reply:
`handleAttributeAssignment` is called in two paths:
1. treat attribute as type, which doesn't change the data set
2. as a handler called in response to `GraphModel.setAttributeID()`, in which case the necessary changes to the `DataConfigurationModel` have already occurred. It may be that the logic within the `handleAttributeAssignment()` method will need to change, but I don't have a clear picture of what those changes look like so I'm inclined to leave them for a subsequent PR.

Reply:
Agreed.